### PR TITLE
Splitting up #921: Immune-deconv changes

### LINKED ISSUE #926: Updated analysis: GSEA throws error with stranded collapse-rnaseq v18 data
#### What analysis module should be updated and why?

I was working on updating the histology color palette for the final versions of the figures (#898) and was having trouble getting the gsea part of transcriptomic-overview figure to run after I had re-ran `collapse-rnaseq` that makes the input files for GSEA. 

How to duplicate this error: 1) be on the master branch with v18 data, run `bash collapse-rnaseq/run-collapse-rnaseq.sh` and then try to run `bash analyses/gene-set-enrichment-analysis/run-gsea.sh` (Or just run the stranded step). 

Something is not lining up as indicated by the `dimnames`. This error comes specifically when the stranded data from collapse-rnaseq from v18 data is run (poly-a runs without error) and [when the actual `GSVA::gsea()` step is run](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/f85831fe7e405b1c6b6c43274cabcc796b7bd3b1/analyses/gene-set-enrichment-analysis/01-conduct-gsea-analysis.R#L98)

```
Estimating GSVA scores for 50 gene sets.
Computing observed enrichment scores
Estimating ECDFs with Gaussian kernels
Using parallel with 8 cores
  |======================================================================| 100%
Error in dimnames(x) <- dn :
  length of 'dimnames' [1] not equal to array extent
Calls: <Anonymous> ... <Anonymous> -> .local -> .gsva -> rownames<- -> rownames<-
In addition: Warning message:
In mclapp(gset.idx.list, ks_test_m, gene.density = rank.scores,  :
  scheduled core 5 did not deliver a result, all values of the job will be affected
Execution halted
```

I can dig into this more, but wanted to post this in case someone more familiar with the `collapse-rnaseq` and/or `gene-set-enrichment-analysis` has more insight. 

It does not fail in CI, so this has something to do with the full set as opposed to the subset of data used in CI. 

#### What changes need to be made? Please provide enough detail for another participant to make the update.
I haven't totally pinpointed what the problem is. But I can dig into it more. Just wanted to post in case someone knows off the bat. 

#### What input data should be used? Which data were used in the version being updated?
v18 data

#### When do you expect the revised analysis will be completed?
We can have a better idea about this once we have a better idea about the source of the problem.

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?

This PR is a split off of #921 where I am updating the transcriptomic overview plot in response to the histology color palette table added in #898

This PR updates immune-deconv module to use the display group and hex codes of the  `palettes/histology_label_color_table.tsv` table that is created by the notebook added in #899  

#### What was your approach?

This PR updates one of the three modules that are ultimately used in the transcriptomic-overview plot that is ultimately updated in #921. 

`immune_deconv` updates:
- `analyses/immune-deconv/01-immune-deconv.R` need to be updated to use `display_group` instead of `short_histology`.
- Also needed to update `analyses/immune-deconv/02-summary-plots.R` in the same way
- After these changes I re-ran `OPENPBTA_DECONV_METHOD="mcp_counter" ./analyses/immune-deconv/run-immune-deconv.sh` (The CIBERSORT.R Doesn't work unless you have that CIBERSORT.R script which we do not). 

*Note that the `palettes/histology_color_palette.tsv` file itself is not deleted yet, I will do this in the last PR of this series (#927) so the other modules that use this file don't break in CI yet. 

#### What GitHub issue does your pull request address?
Makes more progress toward addressing #898 
Closes #926 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
- I ran it with the DECONV_METHOD used in circle CI and in the generate-figures script, but I notice there were originally plots for this method. Does this mean we don't really want this methods plots? 
- I have more questions about why we are keeping CIBERSORT based methods since we can't re-run this (and haven't for almost a year). (We don't have access to the CIBERSORT.R script needed for this). 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes?

### Results

The results themselves don't really change here but will change in transcriptomic-overview. 

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile. -- no new packages are needed. 
- [x] This analysis has been added to continuous integration. 

#### Documentation Checklist

I checked the respective READMEs for these analyses and they did not require any updates for these changes. 

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date. 
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/immune-deconv/02-summary-plots.R [hunk 1]
```diff
@@ -29,13 +29,9 @@ deconvout <- opt$input
 output <- opt$output
 load(deconvout) 
 
-# if short histology is Medulloblastoma or ATRT, use them as broad histology
```
Comment (cansavvy) [inline analyses/immune-deconv/02-summary-plots.R:32]: I didn't think this was relevant anymore so I deleted this step completely, but someone should let me know if something like this is still needed.

Comment (jharenza) [reply analyses/immune-deconv/02-summary-plots.R:32]: confirmed this is ok to delete with the switch to harmonized dx

**Review by cbethell — COMMENTED** [review]
The changes made to the scripts in this PR look good to me!

It appears that parts of the `heatmap_MCPcounter.pdf` and `heatmap_xCell.pdf` plots have been cutoff so the sizing will likely need to be adjusted.

As for your point in this PR's original comment:
> I have more questions about why we are keeping CIBERSORT based methods since we can't re-run this (and haven't for almost a year). (We don't have access to the CIBERSORT.R script needed for this)

I also had the same question so I took a look at the `README`... 

Per this module's `README.md`, 
> "We chose CIBERSORT (abs.) over MCP-Counter as it has the greatest number of overlapping immune cell types with that of xCell".

Also stated that, 
> "We recommend using MCP-Counter instead of CIBERSORT (abs.) when these files are not available to the user".

That said, I would argue that if this PR in particular goes in as is, we should update the `README.md` to reflect the change in method used.

**Comment by cansavvy** [comment]
> Per this module's `README.md`,
> > "We chose CIBERSORT (abs.) over MCP-Counter as it has the greatest number of overlapping immune cell types with that of xCell".
> > "We recommend using MCP-Counter instead of CIBERSORT (abs.) when these files are not available to the user".

Yes. I saw these recommendations in the README. This is part of my question. Why are we keeping this around if it isn't something we can run? It might be something for a separate issue. 

On the separate note, I did notice the plots are trimmed badly, I'll look into fixing that.

**Comment by cansavvy** [comment]
@cbethell , I tracked the CIBERSORT method issue mentioned above here: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/931 and I fixed the plot margins. Let me know what you think.

**Review by cbethell — APPROVED** [review]
With issue #931 filed, this PR LGTM! 👍

**Pull request merged**